src/town_cmd.cpp

```diff
@@ -518,6 +518,43 @@ static void AdvanceHouseConstruction(TileIndex tile)
 	if (flags & BUILDING_HAS_4_TILES) AdvanceSingleHouseConstruction(TILE_ADDXY(tile, 1, 1));
 }
 
+static void OriginalTownProduceCargo(Town *t, TownProductionEffect tpe, uint8_t rate, StationFinder &stations)
+{
+	for (const CargoSpec *cs : _town_production_cargoes[tpe]) {
+		uint32_t r = Random();
+		if (GB(r, 0, 8) < rate) {
+			CargoID cid = cs->Index();
+			uint amt = GB(r, 0, 8) / 8 + 1;
+
+			if (EconomyIsInRecession()) amt = (amt + 1) >> 1;
+			t->supplied[cid].new_max += amt;
+			t->supplied[cid].new_act += MoveGoodsToStation(cid, amt, SourceType::Town, t->index, stations.GetStations());
+		}
+	}
+}
+
+static void BitcountTownProduceCargo(Town *t, TownProductionEffect tpe, uint8_t rate, StationFinder &stations)
+{
+	for (const CargoSpec *cs : _town_production_cargoes[tpe]) {
+		CargoID cid = cs->Index();
+		uint32_t r = Random();
+
+		/* Make a bitmask with up to 32 bits set, one for each potential pax. */
+		int genmax = (rate + 7) / 8;
+		uint32_t genmask = (genmax >= 32) ? 0xFFFFFFFF : ((1 << genmax) - 1);
+
+		/* Mask random value by potential pax and count number of actual pax. */
+		uint amt = CountBits(r & genmask);
```

Comment:
Wow... what a convoluted way of essentially doing `RandomRange`, though I acknowledge that it's just moved code.
It's something like: `uint amt = RandomRange(1 + std::min(32, (rate + 7) / 8));`, right? The +1 because `RandomRange`'s limit is not included in the resulting values.

Reply:
Must admit I didn't try to understand what it wants to do!

Reply:
Never mind-ish. It is getting a random number is the range, but instead of a even chance there is a binominal spread. In other words, the middle of the range is way more likely than the extremities of the range.
It would have been better if that got wrapped up in a separate function in random_func, but for now I'd say moving it is fine.